### PR TITLE
[CARBONDATA-888] Added include and exclude dictionary columns in dataframe writer

### PR DESCRIPTION
**Pull request by sanoj-mg**
Added options dictionary_include and dictionary_exclude in dataframe writer. 

Tested successfully in spark 2.6 / HDP 2.5

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1521/

**Comment by sanoj-mg** [comment]
@jackylk Can you please have a look at the test case? Not sure if "desc formatted" the right way to get these column attributes of a carbondata table.

**Comment by jackylk** [comment]
@sanoj-mg Can you refer to #737, I am doing similar thing for dataframe writer in spark2 module

**Comment by chenliang613** [comment]
@sanoj-mg please let me know your JIRA account's emailid, i will give your contributor right, then you can assign issues to yourself.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1523/

**Comment by sanoj-mg** [comment]
@jackylk I have changed test cases. Can you please review this pr.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1541/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1544/

File: integration/spark/src/main/scala/org/apache/carbondata/spark/CarbonDataFrameWriter.scala [hunk 1]
```diff
@@ -175,13 +175,17 @@ class CarbonDataFrameWriter(val dataFrame: DataFrame) {
   }
 
   private def makeCreateTableString(schema: StructType, options: CarbonOption): String = {
+    val properties = Map("DICTIONARY_INCLUDE" -> options.dictionaryInclude,
+      "DICTIONARY_EXCLUDE" -> options.dictionaryExclude).filter(_._2.isDefined)
+      .map(p => s"'${p._1}' = '${p._2.get}'").mkString(",")
```
Comment (jackylk) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/CarbonDataFrameWriter.scala]: please format it as in spark2 below

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1552/

**Comment by jackylk** [comment]
LGTM, thanks

**Comment by jackylk** [comment]
I will merge it later as we are preparing version 1.1 release, code is freezed for master branch. Or you can create another PR to merge into 12-dev branch. Thanks.

**Comment by sanoj-mg** [comment]
@jackylk I have created another PR #786 to merge into 12-dev branch

**Comment by jackylk** [comment]
#786 is merged, please close this PR

**Pull request closed without merging**